src/main/java/frc/robot/commands/AutoDriveForward.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+
+public class AutoDriveForward extends CommandBase {
+    private final Drivetrain m_drivetrain;
+
+    public AutoDriveForward(Drivetrain subsystem) {
+        m_drivetrain = subsystem;
+        addRequirements(m_drivetrain);
+    }
+    
+    @Override
+    public void initialize() {
+        //Drive forward 3 feet at 20% speed
+        while(m_drivetrain.getEncoderDistance(1) < 36) {   //TODO: put gear ratio
```

Comment:
This will not let the robot do anything else while it is driving. You should not have to write any loops yourself in the code. `initialize()` is run once when the command is first "scheduled". Then, each time the robot main loop is run, `execute()` is called, until `isFinished` returns true. Refer to the command documentation. 

So m_drivetrain.tankDrive should be in the execute method, and not in a loop.